### PR TITLE
Ensure the current post does not appear in auto generated lists of backfilled posts 

### PR DESCRIPTION
**Pull request by attackant**
Addresses #93 

Ensure the current post is not included in the any auto generated lists of backfilled posts (but the user can still choose it.) Also includes `npm audit fix` and a minor refactor for readability (hopefully)

### CODE REVIEW

**Comment by mogmarsh** [comment]
I don't think this addresses the front end

**Comment by mogmarsh** [comment]
I think to handle the front end, we should add a query class that will exclude the current post, then add that query class in https://github.com/alleyinteractive/wp-curate/blob/develop/src/features/class-query-block-context.php
I can try to help, or @dlh01 really knows this setup.

**Comment by dlh01** [comment]
The `Exclude_Queries` class in Type Extensions could be used, if you can supply the current post ID to it: https://github.com/alleyinteractive/wp-type-extensions/blob/a67070fc2d03422c5e24f83a211bffea7845ffa3/src/alley/wp/post-queries/class-exclude-queries.php

**Comment by attackant** [comment]
> The `Exclude_Queries` class in Type Extensions could be used, if you can supply the current post ID to it: https://github.com/alleyinteractive/wp-type-extensions/blob/a67070fc2d03422c5e24f83a211bffea7845ffa3/src/alley/wp/post-queries/class-exclude-queries.php

Thanks! I will check it out, I was just trying to roll my own

File: blocks/query/edit.tsx [hunk 1]
```diff
@@ -169,21 +170,31 @@ export default function Edit({
...
+      apiFetch({ path }).then((response:any) => {
+        let revisedResponse;
+        if (Array.isArray(response)) {
+          revisedResponse = response.filter((item) => item !== currentPostId);
+        } else if (response.id === currentPostId) {
+          // If the current post is in the response, remove it.
+          revisedResponse = null;
+        } else {
+          revisedResponse = response;
+        }
```
Comment (mogmarsh) [inline blocks/query/edit.tsx]: the response should always be an array. I'm not sure how to convince typescript of that.

**Review by mogmarsh — APPROVED** [review]
I think this looks good. As far as I can tell the logic should be the same on the front end. Did you get the same results in the editor and the front end with this change? And did you get the same results on the front end with the current production and this PR (with the exception of removing the current post)?
I'm going to create an issue to add some tests so we can make changes like this with more confidence.

File: src/features/class-query-block-context.php [hunk 2]
```diff
@@ -20,99 +20,127 @@
...
+use Alley\WP\WP_Curate\Trending_Post_Queries;
 use WP_Block;
 use WP_Block_Type;
 use WP_Block_Type_Registry;
 
 /**
  * Provides context to query blocks
  */
-final class Query_Block_Context implements Feature {
+final readonly class Query_Block_Context implements Feature {
```
Comment (dlh01) [inline src/features/class-query-block-context.php]: As the CI results already noted, readonly classes aren't available while the plugin supports PHP 8.1.

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,10 @@
 
 All notable changes to `WP Curate` will be documented in this file.
 
+## 1.8.3 - 2024-04-04
+
+- Enhancement: Exclude current post in backfilled posts query.
```
Comment (dlh01) [inline CHANGELOG.md:7]: Since there are no bugfixes in the release, it seems more like a minor release than a patch release to me, i.e. `1.9.0`.

**Review by dlh01 — COMMENTED** [review]
Updates on the queries side look good to me!

**Pull request merged**